src/drivers/angular2.js

```diff
@@ -0,0 +1,99 @@
+        {
+            Component: window.ng.core.Component,
+            NgModule: window.ng.core.NgModule,
+            ElementRef: window.ng.core.ElementRef,
+            NgZone: window.ng.core.NgZone,
+            BrowserModule: window.ng.platformBrowser.BrowserModule
+        } 
+        : false;
+    },
+
+    register(xcomponent, { Component, NgModule, ElementRef, NgZone, BrowserModule }) {
+
+        xcomponent.log('initializing angular2 component');
+        
+        const getBindingMetadata = () => {
+            const inputs = [];
+            if (xcomponent.looseProps) {
+                inputs.push('props');
+            } else {
+                for (let key of Object.keys(xcomponent.props)) {
+                    inputs.push(key);
+                }
+            }
+            return { inputs };
+        };
+
+
+        function getProps(component) {
+            const props = component.props || omit(component, ['$xContext']);
+            return replaceObject(props, (value, key, fullKey) => {
+                if (typeof value === 'function') {
+                    return function () {
+                        let result;
+                        component.$xContext.zone.run(() => {
+                            result = value.apply(this, arguments);
+                        });
+                        return result;
+                    };
+                }
+            });
+        }
+
+        const bindingMetadata = getBindingMetadata();
+
+        const Angular2Component = 
+            Component({
+                selector: xcomponent.tag,
+                template: `
+                    <div></div>
+                `,
+                inputs: bindingMetadata.inputs
+            })
+            .Class({
+                constructor: [ElementRef, NgZone, function(elementRef, zone) { 
+                    this.$xContext = {
+                        elementRef,
+                        zone
+                    };
+                }],
+                ngOnInit: function () {
+                    const targetElement = this.$xContext.elementRef.nativeElement;
+                    const parent = xcomponent.init(getProps(this), null, targetElement);
+                    parent.render(targetElement);
+                    this.$xContext.parent = parent;
+                },
+                ngOnChanges: function() {
+                    if (this.$xContext.parent) {
+                        this.$xContext.parent.updateProps(getProps(this));
+                    }
+                }
+            });
+
+
+        const Angular2Module = NgModule({
+            imports: [BrowserModule],
+            declarations: [Angular2Component],
+            exports: [Angular2Component]
+        })
+        .Class({
+            constructor: function () {
+                
+            }
+        });
+        
+        xcomponent.driverResults.angular2 = {
+            module: Angular2Module
+        };
+
+        return xcomponent;
```

Comment:
OK I just tried this, and it seemed to work pretty well:

```javascript
      var HelloComponent = ng.core.Component({
          selector: 'hello-world',
          template: `
              <b>hello world!</b>
          `
      }).Class({
          constructor: function() {}
      });

      var AppComponent = ng.core.Component({
          selector: 'my-app',
          template: `
              <h1>My App</h1>
              <hello-world></hello-world>
          `
      }).Class({
          constructor: function() {}
      });

      var AppModule = ng.core.NgModule({
          imports: [ ng.platformBrowser.BrowserModule ],
          declarations: [ AppComponent, HelloComponent ],
          bootstrap: [ AppComponent ]
      }).Class({
          constructor: function () { }
      });
```

Notice how `HelloComponent` is being imported directly into the `AppModule`. This means we can keep the xcomponent driver a bit simpler and just export the component, rather than having to create a new angular module and export that. Work for you?